app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/FullDetailsFragment.java

```diff
@@ -1510,40 +1523,52 @@ public boolean onMenuItemClick(MenuItem item) {
         }
     };
 
-    private PopupMenu.OnMenuItemClickListener playWithMenuListener = new PopupMenu.OnMenuItemClickListener() {
+    private final PopupMenu.OnMenuItemClickListener playWithMenuListener = new PopupMenu.OnMenuItemClickListener() {
         @Override
         public boolean onMenuItemClick(MenuItem item) {
-            switch (item.getItemId()) {
+            return playWithItemMenu(item, false);
+        }
+    };
 
-                case R.id.play_with_vlc:
-                    systemPreferences.getValue().set(SystemPreferences.Companion.getChosenPlayer(),PreferredVideoPlayer.VLC);
-                    play(mBaseItem, 0, false);
-                    return true;
-                case R.id.play_with_exo:
-                    systemPreferences.getValue().set(SystemPreferences.Companion.getChosenPlayer(),PreferredVideoPlayer.EXOPLAYER);
-                    play(mBaseItem, 0, false);
-                    return true;
-                case R.id.play_with_external:
-                    systemPreferences.getValue().set(SystemPreferences.Companion.getChosenPlayer(),PreferredVideoPlayer.EXTERNAL);
-                    PlaybackHelper.getItemsToPlay(ModelCompat.asSdk(mBaseItem), false , false, new LifecycleAwareResponse<List<org.jellyfin.sdk.model.api.BaseItemDto>>(getLifecycle()) {
-                        @Override
-                        public void onResponse(List<org.jellyfin.sdk.model.api.BaseItemDto> response) {
-                            if (!getActive()) return;
+    private final PopupMenu.OnMenuItemClickListener shuffleWithMenuListener = new PopupMenu.OnMenuItemClickListener() {
+        @Override
+        public boolean onMenuItemClick(MenuItem item) {
+            return playWithItemMenu(item, true);
+        }
+    };
 
-                            if (ModelCompat.asSdk(mBaseItem).getType() == BaseItemKind.MUSIC_ARTIST) {
-                                mediaManager.getValue().playNow(requireContext(), response, false);
-                            } else {
-                                videoQueueManager.getValue().setCurrentVideoQueue(response);
-                                navigationRepository.getValue().navigate(Destinations.INSTANCE.externalPlayer(0));
-                            }
-                        }
-                    });
-                    return true;
+    private boolean playWithItemMenu(MenuItem item, boolean shuffle) {
+            if(item.getItemId() == R.id.play_with_vlc) {
```

Comment:
Either keep the switch or use else if statements. Also make sure indenting is correct (space after if)

Reply:
Ok! Fixed to else if. Reason was because android linting apparently does not like switch-cases (i do not really know the details why. But adding code that creates warnings gives me the heebejeebies).